lib/internal/util.js

```diff
@@ -326,14 +326,14 @@ promisify.custom = kCustomPromisifiedSymbol;
 // The build-in Array#join is slower in v8 6.0
 function join(output, separator) {
   let str = '';
-  if (output.length !== 0) {
-    const lastIndex = output.length - 1;
-    for (let i = 0; i < lastIndex; i++) {
+  if (output.length) {
+    const arrayLen = output.length;
+    // Using `var` is much more faster than `let`
+    for (var i = 0; i < arrayLen; i++) {
       // It is faster not to use a template string here
       str += output[i];
-      str += separator;
+      str += (i !== arrayLen - 1) ? separator : '';
```

Comment:
This looks costly, which may be shown by a quick benchmark or simply `time`-ing this for a large number of iterations.

Reply:
I think V8 should do the optimization to inline the stuff and remove comparison as the comparison is already provided. Will do a timing benchmark.